### PR TITLE
Add nonroot user to dockerfile

### PR DESCRIPTION
**Pull request by anna-cross**
# Description

Make sure that container running the Function image is nonroot.

Fixes # https://github.com/meroxa/product/issues/415

## Type of change

Please delete options that are not relevant.

- [ ] New feature
- [ ] Bug fix
- [x] Refactor
- [ ] Documentation

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Tests
- [x] Manual Tests
- [ ] Deployed to staging

### CODE REVIEW

File: src/function-deploy/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,13 @@
+
+FROM node:16-alpine as 
+
+#Set user permissions to nonroot 
+RUN groupadd --gid 1000 nonroot \
```
Comment (lyuboxa) [inline src/function-deploy/Dockerfile:5]: there should always be a nobody/nobody at 1000

Comment (lyuboxa) [reply src/function-deploy/Dockerfile:5]: Sorry, I was wrong:
```
klou=; docker run -it --rm --entrypoint /bin/sh node:16-alpine 
/ # cat /etc/passwd
[..]
nobody:x:65534:65534:nobody:/:/sbin/nologin
node:x:1000:1000:Linux User,,,:/home/node:/bin/sh
```

**Review by lyuboxa — APPROVED** [review]

File: src/function-deploy/Dockerfile [hunk 2]
```diff
@@ -21,4 +23,7 @@ COPY --from=DATA_APP_BUILDER ./app/data-app .
 WORKDIR /app/function-app
 COPY --from=FUNCTION_APP_BUILDER ./app/function-app .
 
+#Set user permissions to nonroot 
+USER nobody
```
Comment (jayjayjpg) [inline src/function-deploy/Dockerfile:27]: Neat, thank you for the swift fix! :sparkles:

**Review by jayjayjpg — APPROVED** [review]
:sparkles:

**Pull request merged**